### DISCUSSION TITLE
linkIdentity doesn't fail if the candidate identity is already linked to another user.

### DISCUSSION THREAD

**Discussion by f-bog** [supabase/supabase#27061] [Questions] [answered]
Hi everyone,

I'm working on converting anonymous users to permanent ones using linkIdentity. It works fine when the provider account hasn't been linked to a user before. However, if I try to link an anonymous user to an already linked provider account, I would expect an error. But nothing happens.

I'm using Expo and the expo-web-browser to handle these authentication scenarios. When attempting to link, the browser opens for a split second before dismissing. Nothing is returned in the error object, and there's no way to identify the error to display a meaningful message on the client.

According to the [documentation](https://supabase.com/docs/reference/javascript/auth-linkidentity), this should fail. Am I missing something?

```js
 const { data, error } = await supabase.auth.linkIdentity({
          provider: "google",
          options: {
            redirectTo,
          },
        });

 if (error) throw error; // error is null here

 const res = await WebBrowser.openBrowserAsync(data.url); // prompts open browser for a split second
```

**Comment by f-bog** [comment]
Should I move this over to bugs?

**Comment by oberbeck** [comment]
Same for me using the Flutter SDK.

**Comment by mattrltrent** [comment]
Same here. Serious issue. Any news on a fix?

**Comment by anthdm** [comment]
Serious issue. I don't even understand this gets no prior at all.

**Comment by OctaviPrats** [comment, marked as answer]
Hey everyone, I ran into the same issue and finally figured out what’s going on — sharing the solution to save others some time:

When using linkIdentity(), if the identity you're trying to link is already associated with another user, the call doesn’t throw an error directly in your code. Instead, Supabase handles this case by redirecting your app to a deep link with the error included in the URL.

So if you're not actively listening to that deep link and checking its contents, you'll miss the error entirely and be left wondering why the link didn’t work.

The fix is to listen to the deep link your app receives after redirection — that’s where Supabase provides the error info (like “identity is already linked to another user”). Once you parse that, you can handle the error gracefully in your app.

Hope this helps someone save hours of debugging

**Comment by marcospgp** [comment]
Does anyone know how to force an identity to be linked? As in, if another user has already linked that identity, unlink it forcefully (since we know current user owns that identity through oauth)
